web/src/compose_send_menu_popover.js

```diff
@@ -118,13 +127,19 @@ export function open_send_later_menu() {
     });
 }
 
-export function do_schedule_message(send_at_time) {
+export function do_schedule_message(send_at_time, remind_message_id) {
     modals.close_if_open("send_later_modal");
 
     if (!Number.isInteger(send_at_time)) {
         // Convert to timestamp if this is not a timestamp.
         send_at_time = Math.floor(Date.parse(send_at_time) / 1000);
     }
+
+    if (remind_message_id !== undefined) {
+        message_reminder.set_message_reminder(send_at_time, remind_message_id);
+        return;
+    }
+
     scheduled_messages.set_selected_schedule_timestamp(send_at_time);
     compose.finish(true);
 }
```

Comment:
I think making it a separate `do_schedule_reminder` function would be more readable -- having the "is this a scheduled message or reminder" detail depend on that second parameter is a bit of a messy abstraction.

Reply:
(I think you'd just have an if/else in the block that chooses that callback? IDK, I could be wrong about what is least messy)